driver/src/main/java/org/neo4j/driver/internal/connector/socket/TrustOnFirstUseTrustManager.java

```diff
@@ -107,12 +110,38 @@ private void saveTrustedHost( String fingerprint ) throws IOException
         logger.warn( "Adding %s as known and trusted certificate for %s.", fingerprint, serverId );
         createKnownCertFileIfNotExists();
 
+        assertKnownHostFileWritable();
         BufferedWriter writer = new BufferedWriter( new FileWriter( knownHosts, true ) );
         writer.write( serverId + " " + this.fingerprint );
         writer.newLine();
         writer.close();
     }
 
+
+    private void assertKnownHostFileReadable() throws IOException
+    {
+        if( !knownHosts.canRead() )
+        {
+            throw new IOException( format(
+                    "Failed to load certificates from file %s as you have no read permissions to it. " +
+                    "Try configuring the Neo4j driver to use a file system location you do have read permissions to.",
+                    knownHosts.getAbsolutePath()
+            ) );
+        }
+    }
+
+    private void assertKnownHostFileWritable() throws IOException
+    {
+        if( !knownHosts.canWrite() )
+        {
+            throw new IOException( format(
+                    "Failed to write certificates to file %s as you have no write permissions to it. " +
```

Comment:
Would appreciate if you could add a `\n` after the first sentence in these messages, just like the ones below for CertificateException.